### DISCUSSION TITLE
Hot Reload XAML and C# in Rider on Mac

### DISCUSSION THREAD

**Discussion by mrc-eu** [AvaloniaUI/Avalonia#14343] [Q&A]
Does Hot Reload XAML and C# work in Rider _on Mac_?

**Comment by thevortexcloud** [comment]
Hot reload of XAML does not work anywhere. It might be added in future as a paid feature. There is https://github.com/AvaloniaUI/Live.Avalonia but people are saying it does not work any more.

**Reply by mrc-eu** [reply]
I guess that hot reload of C# doesn't work also then?

**Reply by thevortexcloud** [reply]
C# will work if your IDE supports it. `dotnet watch` should also work. I know the hot reload experience with rider on Linux is fairly inconsistent (meaning it seemingly does not trigger 60% of the time) but it does work. Not sure about Mac's, I don't use them. Visual Studio on Windows works fine.
